### PR TITLE
Added examples folder and examples file

### PR DESCRIPTION
**Pull request by mrmcwethy**
#https://github.com/adafruit/circuitpython/issues/493

These examples work with the currently released library.

Added examples files for frame.py, blink.py, simple.py, text.py, wave.py
Also linted project.

### CODE REVIEW

**Review by tannewt — APPROVED** [review]
Looks great! Thanks @mrmcwethy !

File: adafruit_is31fl3731.py [hunk 1]
```diff
@@ -1,13 +1,48 @@
-# CircuitPython compatible version of the micropython-adafruit-is31fl3731
-# Charlieplex LED driver:
-#  https://github.com/adafruit/micropython-adafruit-is31fl3731
-# Original micropython-adafruit-is31fl3731 author:
-#   Radomir Dopieralski
-# Port to CircuitPython API:
-#   Tony DiCola
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Toni DiCola
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: Tony

File: adafruit_is31fl3731.py [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+====================================================
+
+CircuitPython driver for the IS31FL3731 charlieplex IC.
+
+This driver supports the following hardware:
+
+* `Adafruit 16x9 Charlieplexed PWM LED Matrix Driver - IS31FL3731 <https://www.adafruit.com/product/2946>`_
+* `Adafruit 15x7 CharliePlex LED Matrix Display FeatherWings <https://www.adafruit.com/product/2965>`_
+
+* Author(s): Toni DiCola
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: Tony

File: adafruit_is31fl3731.py [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
+"""
+#pylint: enable-msg=line-too-long
+
+# imports
 import math
 import time
+from micropython import const
 
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_IS31FL3731.git"
```
Comment (deshipu) [inline adafruit_is31fl3731.py:45]: You can break long strings just like in C:

```
__repo__ = ("https://github.com/adafruit/"
            "Adafruit_CircuitPython_IS31FL3731.git")
```
and then you don't have to disable the line too long warning.

File: adafruit_is31fl3731.py [hunk 4]
```diff
@@ -136,19 +193,27 @@ def fade(self, fade_in=None, fade_out=None, pause=0):
...
+            :param show: True to show the frame; False to don't force a show
+        """
         if frame is None:
             return self._frame
         if not 0 <= frame <= 8:
             raise ValueError("Frame out of range")
         self._frame = frame
         if show:
-            self._register(_CONFIG_BANK, _FRAME_REGISTER, frame);
+            self._register(_CONFIG_BANK, _FRAME_REGISTER, frame)
```
Comment (deshipu) [inline adafruit_is31fl3731.py:210]: Add `return None` here at the end to get rid of the "inconsistent return statement" warning.

File: adafruit_is31fl3731.py [hunk 5]
```diff
@@ -173,6 +240,11 @@ def blink(self, rate=None):
         self._register(_CONFIG_BANK, _BLINK_REGISTER, rate & 0x07 | 0x08)
```
Comment (deshipu) [inline adafruit_is31fl3731.py:245]: ditto

File: adafruit_is31fl3731.py [hunk 6]
```diff
@@ -193,15 +265,25 @@ def fill(self, color=None, blink=None, frame=None):
...
     def pixel(self, x, y, color=None, blink=None, frame=None):
+        """Set blink or brightness for an x,y pixel
+            :param x: horizontal pixel position
+            :param y: vertical pixel position
+            :param color: brightness value 0->255
+            :param blink: True to blink
+            :param frame: the frame to set the pixel
+        """
         if not 0 <= x <= self.width:
             return
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: Change this to `return None` to get rid of the "inconsistent return statement" warning

File: adafruit_is31fl3731.py [hunk 7]
```diff
@@ -193,15 +265,25 @@ def fill(self, color=None, blink=None, frame=None):
...
         if not 0 <= y <= self.height:
             return
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: ditto

File: adafruit_is31fl3731.py [hunk 8]
```diff
@@ -218,13 +300,18 @@ def pixel(self, x, y, color=None, blink=None, frame=None):
             else:
                 bits &= ~(1 << bit)
             self._register(frame, _BLINK_OFFSET + addr, bits)
```
Comment (deshipu) [inline adafruit_is31fl3731.py:309]: add `return None`

File: adafruit_is31fl3731.py [hunk 9]
```diff
@@ -193,15 +265,25 @@ def fill(self, color=None, blink=None, frame=None):
             for col in range(18):
                 self._register(frame, _BLINK_OFFSET + col, data)
 
-    def _pixel_addr(self, x, y):
+    def pixel_addr(self, x, y): #pylint: disable-msg=no-self-use
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: you can use `@staticmethod` here and drop the `self`

File: adafruit_is31fl3731.py [hunk 10]
```diff
@@ -218,13 +300,18 @@ def pixel(self, x, y, color=None, blink=None, frame=None):
...
 
 
 class CharlieWing(Matrix):
+    """Supports the Charlieplexed feather wing
+    """
     width = 15
     height = 7
 
-    def _pixel_addr(self, x, y):
+    def pixel_addr(self, x, y): #pylint: disable-msg=no-self-use
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: also `@staticmethod`

**Review by deshipu — CHANGES_REQUESTED** [review]

**Comment by deshipu** [comment]
Thank you! It looks good, except for some nitpicks, and some possible fixes for the linter.

File: adafruit_is31fl3731.py (hunk 1 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 2 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 3 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py:45]: pylint was complaining about the module description line length
```
* `Adafruit 16x9 Charlieplexed PWM LED Matrix Driver - IS31FL3731 <https://www.adafruit.com/product/2946>`_
* `Adafruit 15x7 CharliePlex LED Matrix Display FeatherWings <https://www.adafruit.com/product/2965>`_
```
I didn't know how to split a quoted doc string, i decided to allow long lines in this comment area.

File: adafruit_is31fl3731.py (hunk 4 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py:210]: ok

File: adafruit_is31fl3731.py (hunk 5 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py:245]: ok

File: adafruit_is31fl3731.py (hunk 9 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 6 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 7 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 8 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py:309]: ok

File: adafruit_is31fl3731.py (hunk 10 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 3 above)
Comment (deshipu) [reply adafruit_is31fl3731.py:45]: You can just put a newline in place of any space, and indent the new line the same as the previous one.

**Comment by mrmcwethy** [comment]
@deshipu if you like my changes, can you merge it?

File: adafruit_is31fl3731.py [hunk 11]
```diff
@@ -67,24 +108,28 @@ def _i2c_write_reg(self, reg, data):
             self.i2c.writeto(self.address, buf)
         finally:
             self.i2c.unlock()
+        return None
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: This one is not needed, since there is no return in that function.

File: adafruit_is31fl3731.py [hunk 12]
```diff
@@ -115,8 +169,17 @@ def autoplay(self, delay=0, loops=0, frames=0):
         self._register(_CONFIG_BANK, _AUTOPLAY1_REGISTER, loops << 4 | frames)
         self._register(_CONFIG_BANK, _AUTOPLAY2_REGISTER, delay % 64)
         self._mode(_AUTOPLAY_MODE | self._frame)
+        return
+
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: This not is not needed either, since the return is empty.

File: adafruit_is31fl3731.py [hunk 13]
```diff
@@ -162,17 +234,26 @@ def audio_play(self, sample_rate, audio_gain=0,
         self._register(_CONFIG_BANK, _GAIN_REGISTER,
                        bool(agc_enable) << 3 | bool(agc_fast) << 4 | audio_gain)
         self._mode(_AUDIOPLAY_MODE)
+        return
```
Comment (deshipu) [inline adafruit_is31fl3731.py]: This one is not needed too.

File: adafruit_is31fl3731.py (hunk 11 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: yes, oops

File: adafruit_is31fl3731.py (hunk 12 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

File: adafruit_is31fl3731.py (hunk 13 above)
Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: ok

Comment (mrmcwethy) [reply adafruit_is31fl3731.py]: @deshipu ok, one more time.

**Review by deshipu — APPROVED** [review]

**Comment by deshipu** [comment]
Thank you! Sorry for all the nitpicking.

**Comment by mrmcwethy** [comment]
No reason to be sorry, its better with your comment.  i should have done the return stuff instead of the #pylint: disable-msg

**Pull request merged**